pkg/shell/index.js

```diff
@@ -245,8 +245,16 @@ define([
              * to know when it can switch into the frame and inject
              * its own additions.
              */
-
             $(frame).on("load", function() {
+                /*
+                 * if the frame is reloaded directly
+                 * unregister and reregister to ensure
+                 * fresh channels
+                 */
+                $(this.contentWindow).on("unload", function () {
+                    router.unregister(frame.contentWindow, address);
+                    router.register(frame.contentWindow);
```

Comment:
Hmmm, I think it would be simpler to have a Router method to just "bump" the channel seed for a window.

Then on "unload" we would call that so the next "init" message in the "message" event handler gets replied to with the new seed.